### ISSUE TITLE
Accessibility of button menu labels and state

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14148]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Two issues exist with nearly all overflow button menus:
- The vast majority are just labeled "Menu", which is not very descriptive.  It's also redundant with the icon and with the value of `aria-haspopup` from #12758, so a screen reader user would hear something like "Menu button has popup menu".
- The expanded state of the menu it controls should be available to the button through `aria-expanded`.  Unfortunately, this is not supported by the Material components (see material-components/material-web#3245).

### Describe the behavior you expected

- Labels should be more descriptive of the menu and remove the redundancy, such as "Automation Options", "Trigger Options", etc.
- Button that opens the menu should follow the [ARIA menu button pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menubutton/).

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

2022.10.x

### In which browser are you experiencing the issue with?

Any

### Which operating system are you using to run this browser?

Any

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Correct.  I submitted a PR to them which is already in the Material 3 pre-release, but HA hasn't even completely eliminated Polymer components so that may be a while...

**Comment by Nardol** [comment]
Still an issue.

**Comment by Nardol** [comment]
Still relevant.
